### PR TITLE
Full I2C OLED support with different screen sizes/types

### PR DESCRIPTION
**Pull request by noman1228**
I hope this is getting closer to what you guys are expecting. I'm also close to having a full BOM list and PCB design to add. 
ESP-JMT

### CODE REVIEW

File: ESPixelStick/ESPixelStick.ino [hunk 1]
```diff
@@ -493,6 +501,8 @@ void loop()
 */
     FeedWDT ();
 
+
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:504]: No reason for these spaces.

File: ESPixelStick/ESPixelStick.ino [hunk 2]
```diff
@@ -536,14 +548,14 @@ void loop()
         if(abs(now() - ConfigLoadNeeded) > LOAD_CONFIG_DELAY)
         {
             FeedWDT ();
-            LoadConfig ();
+                    LoadConfig ();
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:551]: Restore this line.

File: ESPixelStick/ESPixelStick.ino [hunk 3]
```diff
@@ -536,14 +548,14 @@ void loop()
...
         }
     }
 
     if (ConfigSaveNeeded)
     {
         FeedWDT ();
-        SaveConfig ();
+                SaveConfig ();
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:558]: Restore this line

File: ESPixelStick/src/service/DisplayOLED.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
```
Comment (MartinMueller2003) [inline ESPixelStick/src/service/DisplayOLED.h:1]: This should get a pragma once.

File: platformio.ini [hunk 5]
```diff
@@ -36,7 +36,7 @@ extra_scripts =
     pre:.scripts/pio-version.py
     .scripts/download_fs.py
     post:.scripts/CopyTargets.py
-; build_type = debug
+build_type = debug
```
Comment (MartinMueller2003) [inline platformio.ini:39]: Please put back to release mode

**Comment by MartinMueller2003** [comment]
I assume you have some additional discrete components you are putting on the board.
Also, you have not updated the files in the  ESPixelStick\dist directory so your platform will never been seen in the flashtool.

File: ESPixelStick/src/service/DisplayOLED.cpp [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    display.display();
+    display.startscrollright(0x00,0x0F);
+    logcon(String(F("OLED Loaded")));
+    // DEBUG_END;
+} // Begin
+
+void c_OLED::Update()
+{
+    // DEBUG_START;
+    if (millis() > updateTimer_OLED) // Lets check if OLED needs an update only once every 10 seconds
```
Comment (MartinMueller2003) [inline ESPixelStick/src/service/DisplayOLED.cpp:54]: Using millis for something like this requires you to detect and compensate for wrap around. Using the "now()" function to get seconds makes it so you do not need wrap protection.

File: ESPixelStick/src/service/DisplayOLED.cpp [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        updateTimer_OLED += 10000;
+        display.stopscroll();
+        String currIP = WiFi.localIP().toString();
```
Comment (MartinMueller2003) [inline ESPixelStick/src/service/DisplayOLED.cpp:58]: You should use the Network API GetlocalIP for getting IP address.

File: ESPixelStick/src/service/DisplayOLED.cpp [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        String currHost = WiFi.getHostname();
```
Comment (MartinMueller2003) [inline ESPixelStick/src/service/DisplayOLED.cpp:59]: Use GetHostname for this.

**Review by MartinMueller2003 — CHANGES_REQUESTED** [review]

File: ESPixelStick/ESPixelStick.ino (hunk 3 above)
Comment (noman1228) [reply ESPixelStick/ESPixelStick.ino:558]: Formatter....

File: ESPixelStick/src/service/DisplayOLED.cpp (hunk 7 above)
Comment (noman1228) [reply ESPixelStick/src/service/DisplayOLED.cpp:58]: This also fixed showing invalid IP on the first update after boot.

**Comment by noman1228** [comment]
Still trying to get a reliable and inexpensive vendor on Amazon for caps and fuse holders... It's just so cheap and fast to iterate through designs with JLCPCB and knowing the end result will need them.
 Suggestions for additional discrete?

**Review by MartinMueller2003 — APPROVED** [review]
This looks good. I am good with it.

**Comment by MartinMueller2003** [comment]
Shelby does the actual merges.

**Comment by MartinMueller2003** [comment]
RE Vendors, 
I buy from AliExpress and Ebay when I have plenty of time to wait for parts.
I buy from DigiKey and Mouser if I need things fast.
I have bought from Amazon if I need things tomorrow. They tend to be the most expensive option.

**Comment by noman1228** [comment]
Obviously I'm nowhere near calling myself a coder and even when I thought I was (a wee young boy) GitHub or basically any version control didn't exist. Do I need to do anything else?

**Comment by MartinMueller2003** [comment]
Conflicting files have been resolved. Shelby is the person that controls what goes into main. Your code looks good to me and is now in sync with the latest on main.

**Comment by forkineye** [comment]
It seems to be slightly stale now as there are two conflicts in regards to board definitions. Resolve those and I'll get it merged, thanks.

**Comment by forkineye** [comment]
Looks like `platformio.ini` has some issues, check the CI run.

**Comment by noman1228** [comment]
Must stop attempting to code with my child present....

File: ESPixelStick/ESPixelStick.ino [hunk 9]
```diff
@@ -516,7 +524,9 @@ void loop()
 #ifdef SUPPORT_SENSOR_DS18B20
     SensorDS18B20.Poll();
 #endif // def SUPPORT_SENSOR_DS18B20
-
```
Comment (noman1228) [inline ESPixelStick/ESPixelStick.ino:519]: Add back oled support

**Pull request closed without merging**